src/sentry/models/groupowner.py

```diff
@@ -92,6 +97,52 @@ def owner(self):
 
         return ActorTuple.from_actor_identifier(self.owner_id())
 
+    @classmethod
+    def get_autoassigned_owner_cache_key(self, group_id, project_id, autoassignment_types):
+        if not len(autoassignment_types):
+            raise Exception("Requires the autoassignment types")
+        return f"groupowner_id:{group_id}:{project_id}:{':'.join([str(t) for t in autoassignment_types])}"
+
+    @classmethod
+    def get_autoassigned_owner_cached(cls, group_id, project_id, autoassignment_types):
+        """
+        Cached read access to find the autoassigned GroupOwner.
+        """
+        cache_key = cls.get_autoassigned_owner_cache_key(group_id, project_id, autoassignment_types)
+        issue_owner = cache.get(cache_key)
+        if issue_owner is None:
+            issue_owner = (
+                cls.objects.filter(
+                    group_id=group_id, project_id=project_id, type__in=autoassignment_types
+                )
+                .order_by("type")
+                .first()
+            )
+            if issue_owner is None:
+                issue_owner = False
+            # Store either the GroupOwner if exists or False for no owners
+            cache.set(cache_key, issue_owner, READ_CACHE_DURATION)
+
+        return issue_owner
+
+    @classmethod
+    def invalidate_autoassigned_owner_cache(cls, project_id, autoassignment_types):
+        queryset = Group.objects.filter(
```

Comment:
Let's leave a comment here explaining why this query makes sense. Might not be obvious even to us in a few months.